lib/http_perform.c

```diff
@@ -294,19 +297,32 @@ sf_bool STDCALL http_perform(CURL *curl,
         res = curl_easy_perform(curl);
         /* Check for errors */
         if (res != CURLE_OK) {
-            char msg[1024];
-            if (res == CURLE_SSL_CACERT_BADFILE) {
+          if (res == CURLE_COULDNT_CONNECT && curl_retry_ctx->retry_count <
+                                              retry_on_curle_couldnt_connect_count)
+            {
+              retry = SF_BOOLEAN_TRUE;
+              uint32 next_sleep_in_secs = retry_ctx_next_sleep(curl_retry_ctx);
+              log_error(
+                      "curl_easy_perform() failed connecting to server on attempt %d, "
+                      "will retry after %d second",
+                      curl_retry_ctx->retry_count,
+                      next_sleep_in_secs);
+              sleep(next_sleep_in_secs);
```

Comment:
libsnowflakeclient builds on windows/mac/linux 
sleep may not work on winodws. 
You can try this 
#ifdef _WIN32
            Sleep(sleepTime);  // Sleep for sleepTime milli seconds (Sleep(<time in milliseconds>) in windows)
#else
            std::this_thread::sleep_for(std::chrono::milliseconds (std::chrono::milliseconds(sleepTime)));
or you can use the sleep(number of seconds) API as it works on mac/linux
#endif